prototype_source/quantization_in_pytorch_2_0_export_tutorial.rst

```diff
@@ -0,0 +1,350 @@
+                                |            or X86InductorQuantizer,
+                                |            or <Other Backend Quantizer>
+                                |                /
+    —--------------------------------------------------------
+    |                 prepare_pt2e_quantizer                |
+    —--------------------------------------------------------
+                                |
+                         Calibrate/Train
+                                |
+    —--------------------------------------------------------
+    |                      convert_pt2e                     |
+    —--------------------------------------------------------
+                                |
+                    Reference Quantized Model
+                                |
+    —--------------------------------------------------------
+    |                        Lowering                       |
+    —--------------------------------------------------------
+                                |
+            Executorch, or Inductor, or <Other Backends>
+
+Note: ``prepare_pt2e_quantizer`` will be updated to ``prepare_pt2e`` soon.
+
+An existing quantizer object defined for QNNPack/XNNPack is in
+`QNNPackQuantizer <https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py>`__.
+Taking QNNPackQuantizer as an example, the overall Quantization 2.0 flow could be:
+
+::
+
+    import torch
+    import torch._dynamo as torchdynamo
+    from torch.ao.quantization._quantize_pt2e import convert_pt2e, prepare_pt2e
+    import torch.ao.quantization._pt2e.quantizer.qnnpack_quantizer as qq
+
+    class M(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.linear = torch.nn.Linear(5, 10)
+
+        def forward(self, x):
+            return self.linear(x)
+
+    example_inputs = (torch.randn(1, 5),)
+    model = M().eval()
+
+    # Step 1: Trace the model into an FX graph of flattened ATen operators
+    exported_graph_module, guards = torchdynamo.export(
+        model,
+        *copy.deepcopy(example_inputs),
+        aten_graph=True,
+    )
+
+    # Step 2: Insert observers or fake quantize modules
+    quantizer = qq.QNNPackQuantizer()
+    operator_config = qq.get_symmetric_quantization_config(is_per_channel=True)
+    quantizer.set_global(operator_config)
+    prepared_graph_module = prepare_pt2e_quantizer(exported_graph_module, quantizer)
+
+    # Step 3: Quantize the model
+    convered_graph_module = convert_pt2e(prepared_graph_module)
+
+    # Step 4: Lower Reference Quantized Model into the backend
+
+Inside the Quantizer, we will use the ``QuantizationAnnotation API``
+to convey user's intent for what quantization spec to use and how to
+observe certain tensor values in the prepare step. Now, we will have a step-by-step
+tutorial for how to use the ``QuantizationAnnotation API`` with different types of
+``QuantizationSpec``.
```

Comment:
Befor we introduce any of this, we need to explain what annotation API does. For that we first also have to explain quantization spec.

So the flow I imagine would be
```
Quantizer uses annotation API to convey quantization intent for different operators/patterns. Annotation API uses QuantizationSpec to convey intent of how a tensor will be quantized, e.g. dtype, bitwidth, min, max values, symmetric vs. asymmetric etc. See the spec definition at.... Furthermore it also allows quantizer to specify how a tensor value should be observed, e.g. minmax vs. histogram vs. something custom.

Quantization spec is then used to annotate nodes' output tensor or input tensors. Annotating input tensors is equivalent of annotating edge of the graph, while annotating output tensor is equivalent of annotating node. Thus annotation API requires quantizer to annotate nodes (output tensor) or edges (input tensors) of the graph.

For example...
```

Reply:
Thanks for the suggestion. It makes the introduction of ``Quantizer`` and ``QuantizationSpec`` much clearer.